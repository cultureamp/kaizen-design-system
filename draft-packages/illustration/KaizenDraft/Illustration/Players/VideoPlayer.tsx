import React, { useRef, useEffect } from "react"
import classnames from "classnames"
import { IconButton } from "@kaizen/button"

import { assetUrl } from "@kaizen/hosted-assets"
import styles from "../Base.module.scss"
import { canPlayWebm } from "../utils"

import { usePausePlay } from "./usePausePlay"

export type VideoPlayerProps = {
  /**
   * Specifies whether the animation plays as soon as it is rendered.
   * If the user has enabled prefer-reduced-motion their preferences
   * take precedent over this prop.
   */
  autoplay?: boolean

  /**
   * Replay from start when active animation reaches the end of the animation.
   */
  loop?: boolean

  /**
   * Fallback image. Used when rendering of the asset fails, or as a
   * poster for the video player.
   */
  fallback: string

  /**
   * The path of the animation source, excluding the file extension. This
   * Player will preference Webm over mp4.
   */
  source: string

  /**
   * Aspect ratio that is set on the illustration in Scene/Spot which wraps the
   * component in a container, forcing the aspect ratio.
   */
  aspectRatio?: "landscape" | "portrait" | "square"

  onEnded?: () => void
}

export const VideoPlayer = ({
  autoplay = true,
  loop = false,
  fallback,
  source,
  aspectRatio,
  onEnded,
}: VideoPlayerProps): JSX.Element => {
  const videoRef = useRef<HTMLVideoElement>(null)
  const [prefersReducedMotion, setPrefersReducedMotion] =
    React.useState<boolean>(true)

  useEffect(() => {
    /**
     * Setting `muted` on the player is required in Safari/Webkit and older
     * versions of Chome for the video to autoplay (regardless of whether
     * the format contains an audio stream or not).
     *
     * React does not guarentee the `muted` attribute is set on the
     * `video` element. So on load we force set this attribute. See issue:
     * https://github.com/facebook/react/issues/10389
     */
    const { current: videoElement } = videoRef
    if (videoElement !== null) {
      videoElement.setAttribute("muted", "")
    }
  }, [])

  useEffect(() => {
    // when the source of the animation is updated, we need to reload the asset
    // to ensure the video player is in sync with the new source.
    const { current: videoElement } = videoRef
    if (videoElement !== null) {
      videoElement.load()
    }
  }, [source])

  useEffect(() => {
    if (!window) return

    const reducedMotionQuery = window.matchMedia(
      "(prefers-reduced-motion: reduce)"
    )
    setPrefersReducedMotion(reducedMotionQuery.matches)
    const updateMotionPreferences = (): void => {
      const { matches = false } = window.matchMedia(
        "(prefers-reduced-motion: reduce)"
      )
      setPrefersReducedMotion(matches)
    }

    const isLegacyEdge = navigator.userAgent.match(/Edge/)

    const isUnsupportedSafari =
      window.matchMedia("").addEventListener === undefined

    if (isLegacyEdge || isUnsupportedSafari) return

    reducedMotionQuery.addEventListener("change", updateMotionPreferences, true)

    return function cleanup() {
      reducedMotionQuery.removeEventListener("change", updateMotionPreferences)
    }
  }, [])

  useEffect(() => {
    const { current: videoElement } = videoRef
    if (!videoElement) return

    if (prefersReducedMotion) {
      videoElement.pause()
    } else if (autoplay && !prefersReducedMotion) {
      try {
        // Older browsers may not return a promise, so .play could return undefined
        videoElement.play()?.catch(() => {
          /*
           * An DOMException _may_ be raised by some browsers if we
           * programatically interact with the video before the
           * user has interacted with the page. This is okay - so
           * we're going to catch this error without handling it. See:
           * https://developer.mozilla.org/en-US/docs/Web/Media/Autoplay_guide#autoplay_availability
           */
        })
      } catch (e) {
        /**
         * Older browsers will raise a synchronous error because their first implementation
         * of `.play` was not a promise.
         */
      }
    }
    /**
     * Chrome seems to have an issue with changes to autoplay after the video
     * has been mounted. If there is a change in autoplay we need to force the
     * play() method to be called.
     */
  }, [prefersReducedMotion, autoplay])

  useEffect(() => {
    // Add event listeners for the video element
    const { current: videoElement } = videoRef
    if (!videoElement || !onEnded) return

    videoElement.addEventListener("ended", onEnded)

    return function cleanup() {
      videoElement.removeEventListener("ended", onEnded)
    }
  }, [videoRef])

  const pausePlay = usePausePlay(videoRef)

  return (
    <figure
      className={classnames(styles.figure, {
        [`${styles[aspectRatio!]} ${styles.aspectRatioWrapper}`]:
          Boolean(aspectRatio),
      })}
    >
      <video
        muted={true}
        aria-hidden={true}
        preload="metadata"
        ref={videoRef}
        width="100%"
        data-testid="kz-video-player"
        className={styles.wrapper}
        loop={loop}
        poster={assetUrl(`${fallback}.png`)}
        autoPlay={prefersReducedMotion ? false : autoplay}
        playsInline={true}
        tabIndex={-1}
      >
        {/**
         * This seems counter-intuitive, but webm support is codec specific.
         * Only offer webm if we are positive the browser supports it.
         * Reference: https://bugs.webkit.org/show_bug.cgi?id=216652#c1
         */}
        {canPlayWebm() && (
          <source src={assetUrl(`${source}.webm`)} type="video/webm" />
        )}
        <source src={assetUrl(`${source}.mp4`)} type="video/mp4" />
      </video>
      <IconButton
        onClick={(): void => pausePlay.toggle()}
        icon={pausePlay.icon}
        label={pausePlay.label}
        classNameOverride={styles.pausePlayButton}
      />
    </figure>
  )
}
