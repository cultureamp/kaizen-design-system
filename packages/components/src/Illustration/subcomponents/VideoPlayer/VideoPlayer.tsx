import React, { useEffect, useRef } from "react"
import classnames from "classnames"
import { assetUrl } from "@kaizen/hosted-assets"
import { IconButton } from "~components/Button"
import { canPlayWebm } from "../../utils/canPlayWebm"
import { usePausePlay } from "../../utils/usePausePlay"
import styles from "../Base/Base.module.scss"

export type VideoPlayerProps = {
  /**
   * Specifies whether the animation plays as soon as it is rendered.
   * If the user has enabled prefer-reduced-motion their preferences
   * take precedent over this prop.
   */
  autoplay?: boolean

  /**
   * Replay from start when active animation reaches the end of the animation.
   */
  loop?: boolean

  /**
   * Fallback image. Used when rendering of the asset fails, or as a
   * poster for the video player.
   */
  fallback: string

  /**
   * The path of the animation source, excluding the file extension. This
   * Player will preference Webm over mp4.
   */
  source: string

  /**
   * Aspect ratio that is set on the illustration in Scene/Spot which wraps the
   * component in a container, forcing the aspect ratio.
   */
  aspectRatio?: "landscape" | "portrait" | "square"

  onEnded?: () => void
}

export const VideoPlayer = ({
  autoplay = true,
  loop = false,
  fallback,
  source,
  aspectRatio,
  onEnded,
}: VideoPlayerProps): JSX.Element => {
  const videoRef = useRef<HTMLVideoElement>(null)
  const [prefersReducedMotion, setPrefersReducedMotion] =
    React.useState<boolean>(true)
  const [isWebmCompatible, setIsWebmCompatible] = React.useState<boolean>(false)
  const [windowIsAvailable, setWindowIsAvailable] =
    React.useState<boolean>(false)

  useEffect(() => {
    /**
     * Setting `muted` on the player is required in Safari/Webkit and older
     * versions of Chome for the video to autoplay (regardless of whether
     * the format contains an audio stream or not).
     *
     * React does not guarentee the `muted` attribute is set on the
     * `video` element. So on load we force set this attribute. See issue:
     * https://github.com/facebook/react/issues/10389
     */
    const { current: videoElement } = videoRef
    if (videoElement !== null) {
      videoElement.setAttribute("muted", "")
    }
  }, [])

  useEffect(() => {
    // when the source of the animation is updated, we need to reload the asset
    // to ensure the video player is in sync with the new source.
    const { current: videoElement } = videoRef
    if (videoElement !== null) {
      videoElement.load()
    }
  }, [source])

  useEffect(() => {
    if (!window) return

    const reducedMotionQuery = window.matchMedia(
      "(prefers-reduced-motion: reduce)"
    )
    setPrefersReducedMotion(reducedMotionQuery.matches)
    const updateMotionPreferences = (): void => {
      const { matches = false } = window.matchMedia(
        "(prefers-reduced-motion: reduce)"
      )
      setPrefersReducedMotion(matches)
    }

    const isLegacyEdge = navigator.userAgent.match(/Edge/)

    const isUnsupportedSafari =
      window.matchMedia("").addEventListener === undefined

    if (isLegacyEdge || isUnsupportedSafari) return

    reducedMotionQuery.addEventListener("change", updateMotionPreferences, true)

    return function cleanup() {
      reducedMotionQuery.removeEventListener("change", updateMotionPreferences)
    }
  }, [])

  useEffect(() => {
    const { current: videoElement } = videoRef
    if (!videoElement) return

    if (prefersReducedMotion) {
      videoElement.pause()
    } else if (autoplay && !prefersReducedMotion) {
      try {
        // Older browsers may not return a promise, so .play could return undefined
        videoElement.play()?.catch(() => {
          /*
           * An DOMException _may_ be raised by some browsers if we
           * programatically interact with the video before the
           * user has interacted with the page. This is okay - so
           * we're going to catch this error without handling it. See:
           * https://developer.mozilla.org/en-US/docs/Web/Media/Autoplay_guide#autoplay_availability
           */
        })
      } catch (e) {
        /**
         * Older browsers will raise a synchronous error because their first implementation
         * of `.play` was not a promise.
         */
      }
    }
    /**
     * Chrome seems to have an issue with changes to autoplay after the video
     * has been mounted. If there is a change in autoplay we need to force the
     * play() method to be called.
     */
  }, [prefersReducedMotion, autoplay])

  useEffect(() => {
    // Add event listeners for the video element
    const { current: videoElement } = videoRef
    if (!videoElement || !onEnded) return

    videoElement.addEventListener("ended", onEnded)

    return function cleanup() {
      videoElement.removeEventListener("ended", onEnded)
    }
  }, [videoRef])

  const pausePlay = usePausePlay(videoRef)

  useEffect(() => {
    // SSR does not have a window, which is required for for the canPlayWebm helper.
    // Await window render before rendering the component.
    if (windowIsAvailable !== undefined) {
      setIsWebmCompatible(canPlayWebm())
      setWindowIsAvailable(true)
    }
  }, [windowIsAvailable])

  return (
    <figure
      className={classnames(
        styles.figure,
        aspectRatio && styles[aspectRatio],
        aspectRatio && styles.aspectRatioWrapper
      )}
    >
      <video
        muted={true}
        aria-hidden={true}
        preload="metadata"
        ref={videoRef}
        width="100%"
        data-testid="kz-video-player"
        className={styles.wrapper}
        loop={loop}
        poster={assetUrl(`${fallback}.png`)}
        autoPlay={prefersReducedMotion ? false : autoplay}
        playsInline={true}
        tabIndex={-1}
      >
        {isWebmCompatible && (
          <source src={assetUrl(`${source}.webm`)} type="video/webm" />
        )}
        <source src={assetUrl(`${source}.mp4`)} type="video/mp4" />
      </video>
      <IconButton
        onClick={(): void => pausePlay.toggle()}
        icon={pausePlay.icon}
        label={pausePlay.label}
        classNameOverride={styles.pausePlayButton}
      />
    </figure>
  )
}
