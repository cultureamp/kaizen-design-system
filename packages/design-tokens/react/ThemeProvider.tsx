import React from "react"
import { ThemeManager, defaultTheme, Theme } from "../"

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager: ThemeManager
  children: React.ReactNode
}) => {
  const [theme, setTheme] = React.useState<Theme>(
    themeManager.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme) => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManager.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManager.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <ThemeContext.Provider value={theme}>
      {props.children}
    </ThemeContext.Provider>
  )
}

export const useTheme = () => React.useContext(ThemeContext)
