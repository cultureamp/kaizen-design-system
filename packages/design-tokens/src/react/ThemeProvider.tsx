import React, { useState } from "react"
import { ThemeManager, defaultTheme, Theme } from ".."

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
/**
 * @deprecated ThemeProvider is deprecated. Please use KaizenProvider from "@kaizen/components" instead.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager?: ThemeManager
  children: React.ReactNode
}): JSX.Element => {
  const [themeManagerInstance] = useState(
    () => themeManager || new ThemeManager(defaultTheme)
  )

  const [theme, setTheme] = React.useState<Theme>(
    themeManagerInstance.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme): void => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManagerInstance.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManagerInstance.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <>
      <ThemeContext.Provider value={theme}>
        {props.children}
      </ThemeContext.Provider>
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-demi-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-medium.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-regular.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/tiempos/tiempos-headline-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
    </>
  )
}

export const useTheme = (): Theme => React.useContext(ThemeContext)
