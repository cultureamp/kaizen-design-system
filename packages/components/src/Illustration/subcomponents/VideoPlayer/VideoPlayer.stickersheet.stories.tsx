import React from 'react'
import { Meta } from '@storybook/react'
import { StickerSheet, StickerSheetStory } from '~storybook/components/StickerSheet'
import { VideoPlayer } from './index'

export default {
  title: 'Components/Illustrations/VideoPlayer',
  parameters: {
    controls: { disable: true },
    chromatic: { disable: false },
  },
} satisfies Meta

const StickerSheetTemplate: StickerSheetStory = {
  render: ({ isReversed }) => (
    <StickerSheet isReversed={isReversed} title="Aspect Ratio">
      <StickerSheet.Row header="default">
        <div className="border border-dashed border-grey-500">
          <VideoPlayer
            autoplay={false}
            fallback="illustrations/heart/scene/brand-moments-positive-outro"
            source="illustrations/heart/scene/brand-moments-positive-outro"
          />
        </div>
      </StickerSheet.Row>
      <StickerSheet.Row header="landscape">
        <div className="border border-dashed border-grey-500">
          <VideoPlayer
            autoplay={false}
            aspectRatio="landscape"
            fallback="illustrations/heart/scene/brand-moments-positive-outro"
            source="illustrations/heart/scene/brand-moments-positive-outro"
          />
        </div>
      </StickerSheet.Row>
      <StickerSheet.Row header="portrait">
        <div className="border border-dashed border-grey-500">
          <VideoPlayer
            autoplay={false}
            aspectRatio="portrait"
            fallback="illustrations/heart/scene/brand-moments-positive-outro"
            source="illustrations/heart/scene/brand-moments-positive-outro"
          />
        </div>
      </StickerSheet.Row>
      <StickerSheet.Row header="square">
        <div className="border border-dashed border-grey-500">
          <VideoPlayer
            autoplay={false}
            aspectRatio="square"
            fallback="illustrations/heart/scene/brand-moments-positive-outro"
            source="illustrations/heart/scene/brand-moments-positive-outro"
          />
        </div>
      </StickerSheet.Row>
    </StickerSheet>
  ),
}

export const StickerSheetDefault: StickerSheetStory = {
  ...StickerSheetTemplate,
  name: 'Sticker Sheet (Default)',
}
