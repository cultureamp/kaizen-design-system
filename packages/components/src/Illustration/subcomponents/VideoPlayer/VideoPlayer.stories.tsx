import React from "react"
import { Meta } from "@storybook/react"
import { userEvent, within, expect, waitFor } from "@storybook/test"
import {
  StickerSheet,
  StickerSheetStory,
} from "~storybook/components/StickerSheet"
import { VideoPlayer } from "./index"

export default {
  title: "Components/Illustrations/Tests",
  parameters: {
    chromatic: { disable: false },
    controls: { disable: true },
  },
} satisfies Meta

const StickerSheetTemplate: StickerSheetStory = {
  render: ({ isReversed }) => (
    <StickerSheet isReversed={isReversed} title="Aspect Ratio">
      <StickerSheet.Body>
        <StickerSheet.Row rowTitle="default">
          <VideoPlayer
            autoplay={false}
            fallback="illustrations/heart/scene/brand-moments-positive-outro"
            source="illustrations/heart/scene/brand-moments-positive-outro"
          />
        </StickerSheet.Row>
        <StickerSheet.Row rowTitle="landscape">
          <VideoPlayer
            autoplay={false}
            aspectRatio="landscape"
            fallback="illustrations/heart/scene/brand-moments-positive-outro"
            source="illustrations/heart/scene/brand-moments-positive-outro"
          />
        </StickerSheet.Row>
        <StickerSheet.Row rowTitle="portrait">
          <VideoPlayer
            autoplay={false}
            aspectRatio="portrait"
            fallback="illustrations/heart/scene/brand-moments-positive-outro"
            source="illustrations/heart/scene/brand-moments-positive-outro"
          />
        </StickerSheet.Row>
        <StickerSheet.Row rowTitle="square">
          <VideoPlayer
            autoplay={false}
            aspectRatio="square"
            fallback="illustrations/heart/scene/brand-moments-positive-outro"
            source="illustrations/heart/scene/brand-moments-positive-outro"
          />
        </StickerSheet.Row>
      </StickerSheet.Body>
    </StickerSheet>
  ),
}

export const StickerSheetDefault: StickerSheetStory = {
  ...StickerSheetTemplate,
  name: "VideoPlayer: Sticker Sheet",
}

export const PausePlay: StickerSheetStory = {
  render: () => (
    <VideoPlayer
      autoplay={false}
      fallback="illustrations/heart/scene/brand-moments-positive-outro"
      source="illustrations/heart/scene/brand-moments-positive-outro"
    />
  ),
  name: "VideoPlayer: Pause/Play",
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement)
    const element = canvas.getByRole("button")

    waitFor(() => {
      userEvent.click(element)
      expect(element).toHaveAttribute("aria-label", "Pause animation")
    })

    waitFor(() => {
      userEvent.click(element)
      expect(element).toHaveAttribute("aria-label", "Play animation")
    })
  },
}
