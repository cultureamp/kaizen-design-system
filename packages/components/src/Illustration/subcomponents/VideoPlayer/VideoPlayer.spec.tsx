import React from "react"
import { render, screen } from "@testing-library/react"
import { vi } from "vitest"
import { VideoPlayer } from "./VideoPlayer"

const matchMedia = {
  media: "",
  onchange: null,
  addListener: vi.fn(),
  removeListener: vi.fn(),
  addEventListener: vi.fn(),
  removeEventListener: vi.fn(),
}
const mockPrefersReducedMotion = {
  matches: true,
  ...matchMedia,
}
const mockDoesNotPreferReducedMotion = {
  matches: false,
  ...matchMedia,
}
const mockPlay = vi.fn().mockResolvedValue(undefined)
const mockLoad = vi.fn()
const mockPause = vi.fn()

describe("<VideoPlayer />", () => {
  beforeEach(() => {
    window.HTMLMediaElement.prototype.load = mockLoad
    window.HTMLMediaElement.prototype.play = mockPlay
    window.HTMLMediaElement.prototype.pause = mockPause
    window.matchMedia = vi
      .fn()
      .mockImplementation(() => mockDoesNotPreferReducedMotion)
    // this will stop throwing the unstable_flushDiscreteUpdates console error cause by react bug
    // https://stackoverflow.com/a/65338472/18285270
    Object.defineProperty(HTMLMediaElement.prototype, "muted", {
      set: vi.fn(),
    })
  })

  afterEach(() => {
    vi.clearAllMocks()
  })

  it("renders a video player in the document and autoplay", () => {
    render(
      <VideoPlayer
        autoplay
        fallback="illustrations/heart/spot/moods-cautionary.svg"
        source="illustrations/heart/spot/moods-cautionary.webm"
        data-testid="kz-video-player"
      />
    )
    const videoPlayer = screen.getByTestId("kz-video-player")
    expect(videoPlayer).toBeInTheDocument()
    expect(mockPlay).toHaveBeenCalled()
  })

  it("respects the use-reduced-motion preferences of the user", () => {
    window.matchMedia = vi
      .fn()
      .mockImplementation(() => mockPrefersReducedMotion)
    render(
      <VideoPlayer
        autoplay
        fallback="illustrations/heart/spot/moods-cautionary.svg"
        source="illustrations/heart/spot/moods-cautionary.webm"
        data-testid="kz-video-player"
      />
    )
    const videoPlayer = screen.getByTestId("kz-video-player")
    expect(videoPlayer).not.toHaveAttribute("autoplay")
    expect(mockPause).toHaveBeenCalled()
  })

  it("defaults to autoplay when user does not set use-reduced-motion preferences", () => {
    window.matchMedia = vi
      .fn()
      .mockImplementation(() => mockDoesNotPreferReducedMotion)
    render(
      <VideoPlayer
        autoplay
        fallback="illustrations/heart/spot/moods-cautionary.svg"
        source="illustrations/heart/spot/moods-cautionary.webm"
        data-testid="kz-video-player"
      />
    )
    const videoPlayer = screen.getByTestId("kz-video-player")
    expect(videoPlayer).toHaveAttribute("autoplay")
    expect(mockPlay).toHaveBeenCalled()
  })
})
