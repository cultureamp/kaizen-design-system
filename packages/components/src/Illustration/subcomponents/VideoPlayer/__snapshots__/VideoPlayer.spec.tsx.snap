// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoPlayer /> use-reduced-motion defaults to autoplay when user does not set use-reduced-motion preferences 1`] = `
<video
  aria-hidden="true"
  autoplay=""
  class="wrapper"
  data-testid="kz-video-player"
  muted=""
  playsinline=""
  poster="https://d1e7r7b0lb8p4d.cloudfront.net/illustrations/heart/spot/moods-cautionary.svg.png"
  preload="metadata"
  tabindex="-1"
  width="100%"
/>
`;

exports[`<VideoPlayer /> use-reduced-motion respects the use-reduced-motion preferences of the user 1`] = `
<video
  aria-hidden="true"
  class="wrapper"
  data-testid="kz-video-player"
  muted=""
  playsinline=""
  poster="https://d1e7r7b0lb8p4d.cloudfront.net/illustrations/heart/spot/moods-cautionary.svg.png"
  preload="metadata"
  tabindex="-1"
  width="100%"
/>
`;
